lib/facter/operatingsystem/ciscolinux.rb

```diff
@@ -0,0 +1,34 @@
+require 'facter/util/ciscoos'
+require 'facter/operatingsystem/linux'
+
+module Facter
+  module Operatingsystem
+    class CiscoLinux < Linux
+
+      def initialize(file)
+        @cisco_info = Facter::Util::CiscoOS.cisco_release(file)
+      end
+
+      def get_operatingsystem
+        @operatingsystem ||= @cisco_info['ID']
+        @operatingsystem
+      end
+
+      def get_osfamily
+        @osfamily ||= @cisco_info['ID_LIKE']
+        @osfamily
+      end
+
+      def get_operatingsystemrelease
+        @operatingsystemrelease ||= @cisco_info['VERSION']
+        @operatingsystemrelease
+      end
+
+      def get_operatingsystemmajorrelease 
+        if operatingsystemrelease = get_operatingsystemrelease
+          operatingsystemrelease.split("(").first
+        end
+      end
+    end
```

Comment:
Omitting the definition of get_operatingsystemminorrelease lets this fall-through to the Linux implementation at https://github.com/puppetlabs/facter/blob/2.x/lib/facter/operatingsystem/linux.rb#L103. If the version string contains a `.`, that implementation will return a non-nil value. I'm not sure if that's the intent here.